dom.bs

```diff
@@ -2872,6 +2872,135 @@ before a <var>child</var>, with an optional <i>suppress observers flag</i>, run
+<var>newParent</var> before a <a for=/>node</a> <var>child</var>:
+
+<ol>
+ <!-- Start pre-move validity checks -->
+ <li>
+  <p>If <var>newParent</var>'s <a for=/>shadow-including root</a> is not the same as
+  <var>node</var>'s <a for=/>shadow-including root</a>, then <a>throw</a> a
+  "{{HierarchyRequestError!!exception}}" {{DOMException}}.</p>
+
+  <p class=note>This has the side effect of ensuring that a move is only performed if
+  <var>newParent</var>'s <a>connected</a> is <var>node</var>'s <a>connected</a>.</p>
+ </li>
+
+ <li><p>If <var>node</var> is a <a>host-including inclusive ancestor</a> of <var>newParent</var>,
+ then <a>throw</a> a "{{HierarchyRequestError!!exception}}" {{DOMException}}.
+
+ <li><p>If <var>child</var> is non-null and its <a for=tree>parent</a> is not <var>newParent</var>,
+ then <a>throw</a> a "{{NotFoundError!!exception}}" {{DOMException}}.
+
+ <li><p>If <var>node</var> is not an {{Element}} or a {{CharacterData}} <a for=/>node</a>, then
+ <a>throw</a> a "{{HierarchyRequestError!!exception}}" {{DOMException}}.</p></li>
+
+ <li><p>If <var>node</var> is a {{Text}} <a for=/>node</a> and <var>newParent</var> is a
+ <a for=/>document</a>, then <a>throw</a> a "{{HierarchyRequestError!!exception}}" {{DOMException}}.
+
+ <li><p>If <var>newParent</var> is a <a for=/>document</a>, <var>node</var> is an {{Element}}
+ <a for=/>node</a>, and either <var>newParent</var> has an <a for=/>element</a>
+ <a for=tree>child</a>, <var>child</var> is a <a>doctype</a>, or <var>child</var> is non-null and a
+ <a>doctype</a> is <a>following</a> <var>child</var> then <a>throw</a> a
+ "{{HierarchyRequestError!!exception}}" {{DOMException}}.
+
+ <!-- Start removing-related bookkeeping steps -->
+ <li><p>Let <var>oldParent</var> be <var>node</var>'s <a for=tree>parent</a>.
+
+ <li><p><a>Assert</a>: <var>oldParent</var> is non-null.
+
+ <li><p>Run the <a>live range pre-remove steps</a>, given <var>node</var>.
+
+ <li><p>For each {{NodeIterator}} object <var>iterator</var> whose
+ <a for=traversal>root</a>'s <a for=Node>node document</a> is <var>node</var>'s
+ <a for=Node>node document</a>, run the <a><code>NodeIterator</code> pre-remove steps</a> given
+ <var>node</var> and <var>iterator</var>.
+
+ <li><p>Let <var>oldPreviousSibling</var> be <var>node</var>'s <a>previous sibling</a>.
+
+ <li><p>Let <var>oldNextSibling</var> be <var>node</var>'s <a for=tree>next sibling</a>.
+
+ <li><p><a for=set>Remove</a> <var>node</var> from <var>oldParent</var>'s <a for=tree>children</a>.
+
+ <li><p>If <var>node</var> is <a for=slottable>assigned</a>, then run <a>assign slottables</a> for
+ <var>node</var>'s <a>assigned slot</a>.
+
+ <li><p>If <var>oldParent</var>'s <a for=tree>root</a> is a <a for=/>shadow root</a>, and
+ <var>oldParent</var> is a <a>slot</a> whose <a for=slot>assigned nodes</a> is the empty list, then
+ run <a>signal a slot change</a> for <var>oldParent</var>.
```

Comment:
Aren't we missing the steps needed to track the case when slot element itself if moving. In the "remove" algorithm step 14.
And I didn't see tests for moving slots.

Reply:
Apologies for the delay on this. Yeah I think you're right. I've added the missing step in https://github.com/whatwg/dom/pull/1307/commits/87b6acbf12368302a05b22ed349d7bacd89b1507 and a test for this in https://chromium-review.googlesource.com/c/chromium/src/+/6298941 which will be upstreamed soon.